packages/camera/ios/Classes/CameraPlugin.m

```diff
@@ -196,6 +254,116 @@ - (FlutterError *_Nullable)onListenWithArguments:(id _Nullable)arguments
   _eventSink = events;
   return nil;
 }
+- (void)startRecordingVideoAtPath:(NSString *)path result:(FlutterResult)result {
+  if (!_isAudioSetup) {
+    [self setUpCaptureSessionForAudio];
+  }
+
+  if (!_isRecording) {
+    NSLog(@"start video recording...");
+    if (![self setupWriterForPath:path]) {
+      NSLog(@"Setup Writer Failed");
+      return;
+    }
+    [_captureSession stopRunning];
```

Comment:
Can we rewire the outputs here without stopping the capture?  This would avoid jank.

Reply:
Once the inputs/outputs have been changed, the session needs to be restarted to refresh it.

Reply:
However, will make sure it's restarted only when audio inputs have been added to prevent restarting unnecessarily.